catroid/src/androidTest/java/org/catrobat/catroid/test/embroidery/DSTFileGeneratorTest.java

```diff
@@ -0,0 +1,203 @@
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.embroidery;
+
+import android.graphics.PointF;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import com.android.dex.util.FileUtils;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.embroidery.DSTFileGenerator;
+import org.catrobat.catroid.embroidery.EmbroideryList;
+import org.catrobat.catroid.io.StorageOperations;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class DSTFileGeneratorTest {
+	final String projectName = "testProject";
+	DSTFileGenerator fileGenerator;
+	File dstFile;
+	final int[] conversion = {-81, 81, -27, 27, -9, 9, -3, 3, -1, 1};
+
+	@Before
+	public void setUp() {
+		Project project = new Project(InstrumentationRegistry.getTargetContext(), projectName);
+		ProjectManager.getInstance().setCurrentProject(project);
+		fileGenerator = new DSTFileGenerator();
+		dstFile = new File(Constants.CACHE_DIR + "/" + projectName + ".dst");
+	}
+
+	private int getValueForByte(int element) {
+		int mask = 0x200;
+		int value = 0;
+		for (int i = 0; i < conversion.length; i++) {
+			if ((element & (mask >>> i)) != 0) {
+				value += conversion[i];
+			}
+		}
+		return value;
+	}
+
+	@Test
+	public void testConversionTable() {
+		for (int element = 0; element < DSTFileGenerator.CONVERSIONTABLE.length; element++) {
+			int value = getValueForByte(DSTFileGenerator.CONVERSIONTABLE[element]);
+			if (element > 121) {
+				assertEquals((element - 121) * (-1), value);
```

Comment:
it would be easier to understand, and way easier to see whats going wrong if something breaks, if youd also compile a expected and actual list and then compare them with arrayEquals